src/storage/storage.cc

```diff
@@ -149,6 +155,9 @@ void StorageImpl::DirectFree(Storage::Handle handle) {
         return nullptr;
       });
   this->ActivateDevice(ctx);
+#if MXNET_USE_PROFILER
+  profiler_.OnFree(handle);
```

Comment:
shouldn't it be placed after free?

Reply:
I put it before in case Free() throws some sort of exception, it will still remove it. I can put it on the other side if you like.